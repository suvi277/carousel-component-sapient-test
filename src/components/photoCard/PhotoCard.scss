.carousel-item {
    margin: 20px 10px;
    list-style: none;
    width: 100%;
    display: flex;
    flex-direction: column;

    img {
        width: 100%;
        flex: 1 1 auto;
    }

    .user-name {
        padding-top: 10px;
        margin: 0;
        text-align: center;
        font-size: 20px;
    }

    @media screen and (min-width: 768px) {
        width: calc(100% / 3 - 20px);
        text-align: left;

        &:nth-child(2) {
            border: 2px solid #FF5733;

            p {
                text-align: center;
                padding-bottom: 12px;
            }
        }
    }

    @media screen and (min-width: 1024px) {
        width: calc(100% / 5 - 20px);
        text-align: left;

        &:nth-child(3) {
            border: 4px solid #FF5733;

            p {
                text-align: center;
                padding-bottom: 12px;
            }

        }

        &:nth-child(2) {
            border: none;

            p {
                text-align: left;
                padding-bottom: 0;
            }
        }
    }
}
