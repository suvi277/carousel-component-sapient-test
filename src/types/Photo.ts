export interface Photo {
	previewURL: string;
	likes: number;
	user: string;
}

interface ServiceLoading {
	status: string;
}

interface ServiceLoaded<Photo> {
	status: string;
	photos?: Array<Photo>;
}

interface ServiceError {
	status: string;
	error?: Error;
}

export interface BaseAction {
	type: string;
	payload?: any;
}

export type Carousel<T> = ServiceLoading | ServiceLoaded<T> | ServiceError;
