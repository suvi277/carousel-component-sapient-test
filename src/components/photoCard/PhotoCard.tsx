import React from 'react';
import { Photo } from '../../types/Photo';
import './PhotoCard.scss';

const PhotoCard: React.FC<Photo> = ({ previewURL, user }) => {
  return (
    <li className="carousel-item">
      <img className="carousel-image" src={previewURL} alt={user} />
      <p className="user-name">{user}</p>
    </li>
  );
};

export default PhotoCard;
