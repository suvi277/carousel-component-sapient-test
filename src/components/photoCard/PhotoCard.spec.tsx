import React from 'react';
import { shallow, mount } from 'enzyme';
import { Photo } from '../../types/Photo';
import PhotoCard from './PhotoCard';

describe('<PhotoCard /> with default props', () => {
  const testProps = {
    previewURL: 'sample title',
    user: 'sample owner name',
    likes: 12,
  };
  const wrapper = shallow(<PhotoCard {...testProps} />);
  it('renders as expected', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should exist card and shadow-lg classname', () => {
    expect(wrapper.find('.carousel-item')).toHaveLength(1);
  });

  it('should insert the prop values to correct places ', () => {
    const props: Photo = {
        ...testProps,
      },
      photoCardComp = mount(<PhotoCard {...props} />);

    expect(photoCardComp.find('.user-name').last().render().text()).toEqual(
      props.user
    );
    expect(photoCardComp.find('.carousel-image').render().attr('src')).toEqual(
      props.previewURL
    );
  });
});
