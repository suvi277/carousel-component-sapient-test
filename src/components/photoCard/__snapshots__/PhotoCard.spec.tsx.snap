// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PhotoCard /> with default props renders as expected 1`] = `ShallowWrapper {}`;
